commands/files/file.go

```diff
@@ -11,6 +11,57 @@ var (
 	ErrNotReader    = errors.New("This file is a directory, can't use Reader functions")
 )
 
+// An AdvReader is like a Reader but supports getting the current file
+// path and offset into the file when applicable.
+type AdvReader interface {
+	io.Reader
+	ExtraInfo() ExtraInfo
```

Comment:
@whyrusleeping , It is used to pass needed information through the layers into the filestore such as the absolute pathname and the offset into the file.  It is also used to pass addOpts through the lawyers to avoid that extra parameter.  Are you looking for specific spots in the code where the information is used?